rust/connlib/tunnel/src/peer.rs

```diff
@@ -138,6 +139,7 @@ pub struct ClientOnGateway {
     ipv4: Ipv4Addr,
     ipv6: Ipv6Addr,
     resources: HashMap<ResourceId, ResourceOnGateway>,
+    internet_resource_enabled: bool,
```

Comment:
Isn't this implicit from having an Internet Resource in the hash map? Why track this separately?

Reply:
If this is just a cache, let's document that. Plus, should we have a follow-up to make it an either resources or Internet resource in here once Internet Sites are released?

Reply:
Let's do it as part of #6982